File: backend/src/main/java/com/btalk/entity/Notification.java
package com.btalk.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import org.hibernate.annotations.GenericGenerator;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "notifications")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Notification {
    
    @Id
    @GeneratedValue(generator = "uuid2")
    @GenericGenerator(name = "uuid2", strategy = "uuid2")
    @Column(name = "notification_id", columnDefinition = "BINARY(16)")
    private UUID notificationId;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "recipient_id", nullable = false)
    private User recipient;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "sender_id")
    private User sender;
    
    @Column(name = "title", nullable = false)
    private String title;
    
    @Column(name = "message", nullable = false, length = 1000)
    private String message;
    
    @Enumerated(EnumType.STRING)
    @Column(name = "type", nullable = false)
    private NotificationType type;
    
    @Column(name = "data", length = 2000)
    private String data; // JSON data for additional information
    
    @Column(name = "is_read", nullable = false)
    @Builder.Default
    private Boolean isRead = false;
    
    @Column(name = "is_deleted", nullable = false)
    @Builder.Default
    private Boolean isDeleted = false;
    
    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;
    
    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
    
    @Column(name = "read_at")
    private LocalDateTime readAt;
    
    public enum NotificationType {
        NEW_MESSAGE,
        NEW_CONVERSATION,
        MESSAGE_REACTION,
        USER_ONLINE,
        USER_OFFLINE,
        CALL_INCOMING,
        CALL_MISSED,
        CALL_ENDED,
        FILE_SHARED,
        MENTION,
        SYSTEM_ANNOUNCEMENT
    }
} 

File: frontend/src/app/models/notification.ts
export interface Notification {
  notificationId: string;
  recipientId: string;
  senderId?: string;
  senderName?: string;
  senderAvatar?: string;
  title: string;
  message: string;
  type: NotificationType;
  data?: string;
  isRead: boolean;
  createdAt: string;
  readAt?: string;
}

export enum NotificationType {
  NEW_MESSAGE = 'NEW_MESSAGE',
  NEW_CONVERSATION = 'NEW_CONVERSATION',
  MESSAGE_REACTION = 'MESSAGE_REACTION',
  USER_ONLINE = 'USER_ONLINE',
  USER_OFFLINE = 'USER_OFFLINE',
  CALL_INCOMING = 'CALL_INCOMING',
  CALL_MISSED = 'CALL_MISSED',
  CALL_ENDED = 'CALL_ENDED',
  FILE_SHARED = 'FILE_SHARED',
  MENTION = 'MENTION',
  SYSTEM_ANNOUNCEMENT = 'SYSTEM_ANNOUNCEMENT'
}

export interface NotificationRequest {
  recipientId: string;
  senderId?: string;
  title: string;
  message: string;
  type: NotificationType;
  data?: string;
}

export interface NotificationResponse {
  success: boolean;
  message: string;
  data?: Notification;
}

export interface NotificationListResponse {
  success: boolean;
  message: string;
  data?: {
    content: Notification[];
    totalElements: number;
    totalPages: number;
    size: number;
    number: number;
  };
} 